### PR TITLE
[WIP] RWAHS-310 Detail view

### PR DESCRIPTION
**Pull request by leftclickben**
WIP not ready for merge yet.

Includes #6, so please merge that first.

Just putting this up so we can discuss in the morning, I think I know the way forward but want a second opinion before I do it.  Not much work to do, in fact the Library view is functionally complete, and we could do the others the same way, but I don't think it's the best way, there's a lot of duplicated code.

### CODE REVIEW

File: public/app/ui/pages/detail/library.html [hunk 1]
```diff
@@ -1,3 +1,209 @@
 <article id="library-detail" class="detail">
-    <h1>Library Record</h1>
+    <div class="type-header">Library Record</div>
```
Comment (leftclickben) [inline public/app/ui/pages/detail/library.html]: This is the key file.  As you can see there is a lot of logic in the template (`ko if: ...` and `ko foreach: ...`), and it's quite repetitive.  So I'll probably look at knockout components but just wanted to discuss it with you first.

**Comment by kehh** [comment]
There are conflicts and I can't see the actual delta now, could you please merge in upstream/master?

File: public/app/ui/pages/detail/library.html [hunk 2]
```diff
@@ -1,3 +1,209 @@
...
+        <!-- /ko -->
+        <!-- ko ifnot: data().Title -->
+        <span class="text-muted">(untitled)</span>
+        <!-- /ko -->
+    </h1>
+    <div class="row">
+        <div class="col-md-12">
+            <div class="name" data-bind="text: parse(data().Author).length > 1 ? 'Authors' : 'Author'"></div>
+            <div class="value">
+                <!-- ko if: parse(data().Author).length > 0 -->
```
Comment (kehh) [inline public/app/ui/pages/detail/library.html]: Can this become a re-used component?

**Comment by leftclickben** [comment]
Closing in favour of #9

**Pull request closed without merging**